src/main/java/org/example/bookstore/repository/impl/BookRepositoryImpl.java

```diff
@@ -0,0 +1,55 @@
+package org.example.bookstore.repository.impl;
+
+import jakarta.persistence.criteria.CriteriaQuery;
+import java.util.List;
+import org.example.bookstore.model.Book;
+import org.example.bookstore.repository.BookRepository;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public class BookRepositoryImpl implements BookRepository {
+    private final SessionFactory sessionFactory;
+
+    @Autowired
```

Comment:
this annotation is redundant. anyway let's add Lombok and use RequiredArgsConstructor annotation